### PR TITLE
Add dependencies for tools which were missing them

### PR DESCRIPTION
**Pull request by EdwardLarson**
- [x] I have reviewed the [OFRAK contributor guide](https://ofrak.com/docs/contributor-guide/getting-started.html) and attest that this pull request is in accordance with it.

**One sentence summary of this PR (This should go in the CHANGELOG!)**
Mark dependencies on libmagic and `strings` utility.

**Link to Related Issue(s)**
These are tools which OFRAK depends on. On some platforms, like Mac, they are not necessarily installed. They should have been explicitly marked as dependencies so the `ofrak deps` command can find and install them when necessary.

**Please describe the changes in your request.**

**Anyone you think should look at this, specifically?**

### CODE REVIEW

File: ofrak_core/ofrak/core/magic.py [hunk 1]
```diff
@@ -22,18 +21,47 @@ class Magic(ResourceAttributes):
...
+        import magic
+    except ImportError:
+        magic = None
+
+    def __init__(self):
+        super().__init__(
+            "libmagic",
+            "https://linux.die.net/man/3/libmagic",
+            install_check_arg="",
+            brew_package="libmagic",
```
Comment (rbs-jacob) [inline ofrak_core/ofrak/core/magic.py:35]: Should this also have `apt_package="libmagic1"`?

Comment (ANogin) [reply ofrak_core/ofrak/core/magic.py:35]: I think the proper URL is https://www.darwinsys.com/file/. And yes, `apt_package="libmagic1"`

**Review by rbs-jacob — APPROVED** [review]

File: ofrak_patch_maker/ofrak_patch_maker/toolchain/utils.py [hunk 2]
```diff
@@ -5,15 +5,19 @@
...
+_magic = None
+
 
 def get_file_format(path):
-    result = magic.from_file(path).split(" ")[0].lower()
+    global _magic
+    if _magic is None:
+        import magic as _magic
+
+    result = _magic.from_file(path).split(" ")[0].lower()
```
Comment (whyitfor) [inline ofrak_patch_maker/ofrak_patch_maker/toolchain/utils.py:20]: Why is this change needed? I don't understand what is different.

File: ofrak_core/ofrak/core/magic.py [hunk 3]
```diff
@@ -22,19 +22,54 @@ class Magic(ResourceAttributes):
     descriptor: str
 
 
+class _LibmagicDependency(ComponentExternalTool):
+    try:
+        import magic
+
+        magic_: ModuleType = magic
+    except ImportError:
+        magic_ = None
```
Comment (whyitfor) [inline ofrak_core/ofrak/core/magic.py:31]: Instead of duplicating this try-catch logic, why not have a global `MAGIC_INSTALLED` variable like we are already doing with [binwalk](https://github.com/redballoonsecurity/ofrak/blob/master/ofrak_core/ofrak/core/binwalk.py)?

File: ofrak_patch_maker/ofrak_patch_maker/toolchain/utils.py (hunk 2 above)
Comment (EdwardLarson) [reply ofrak_patch_maker/ofrak_patch_maker/toolchain/utils.py:20]: If it is left as a global import, it doesn't matter if a user (without libmagic installed) wants to exclude components missing dependencies because a hard error will be raised at import time when PatchMaker code is imported. The expected behavior is that as a user without an external tool dependency installed, I should still be able to launch and use OFRAK without needing to install all those tools, working instead with a limited set of components.

Since we only use magic in this one place, I thought it acceptable to sort of localize the magic import.

File: ofrak_core/ofrak/core/magic.py (hunk 3 above)
Comment (whyitfor) [reply ofrak_core/ofrak/core/magic.py:31]: @EdwardLarson

File: ofrak_patch_maker/ofrak_patch_maker/toolchain/utils.py (hunk 2 above)
Comment (whyitfor) [inline ofrak_patch_maker/ofrak_patch_maker/toolchain/utils.py:20]: I see. What about deleting lines 12-13, and something like this:

```suggestion
    try:
        import magic
    except ImportError:
        # ImportError is likely raise because libmagic cannot be found on the system. See error message.
        raise
    result = magic.from_file(path).split(" ")[0].lower()
```

File: ofrak_patch_maker/ofrak_patch_maker/toolchain/utils.py [hunk 4]
```diff
@@ -5,15 +5,19 @@
...
 from typing import Optional, Dict, Mapping, Tuple
 
-import magic
-
 from ofrak_patch_maker.toolchain.model import BinFileType, Segment
 from ofrak_type.error import NotFoundError
 from ofrak_type.memory_permissions import MemoryPermissions
 
+_magic = None
+
```
Comment (whyitfor) [inline ofrak_patch_maker/ofrak_patch_maker/toolchain/utils.py:13]: In favor of deleting this in conjunction with https://github.com/redballoonsecurity/ofrak/pull/299/files#r1194275565.:

```suggestion
```

File: ofrak_core/ofrak/core/magic.py [hunk 5]
```diff
@@ -22,19 +28,47 @@ class Magic(ResourceAttributes):
...
     outputs = (Magic,)
+    external_dependencies = (LIBMAGIC_DEP,)
 
     async def analyze(self, resource: Resource, config=None) -> Magic:
         data = await resource.get_data()
-        magic_mime = magic.from_buffer(data, mime=True)
-        magic_description = magic.from_buffer(data)
-        return Magic(magic_mime, magic_description)
+        if not MAGIC_INSTALLED:
+            raise ImportError("libmagic does not seem to be installed!")
```
Comment (whyitfor) [inline ofrak_core/ofrak/core/magic.py:67]: Would it be better to check for (and handle) this error inside of [AbstractResource.run](https://github.com/redballoonsecurity/ofrak/blob/master/ofrak_core/ofrak/component/abstract.py#L100), where other `ComponentMissingDependencyError`s are raised?

**Review by whyitfor — COMMENTED** [review]
Looking good! See one comment about raising a `ComponentMissingDependencyError`.

**Review by whyitfor — APPROVED** [review]

**Comment by whyitfor** [comment]
@EdwardLarson, can you add a "fixed" changelog to OFRAK core for the Magic issue? Otherwise I think this looks good!
